<?xml version="1.0"?>
<launch>
	<arg name="scan_topic" default="scan"/>
		  
	<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
        <remap from="scan" to="$(arg scan_topic)"/>
        
        <param name="odom_frame" value="odom"/>
        <param name="base_frame" value="base_robot"/>
        <param name="map_frame" value="map"/>
        
        <param name="map_update_interval" value="30.0"/> <!--Publish new map-->
        <param name="maxUrange" value="6.0"/>  <!--the maximum usable range of the laser-->
        <param name="maxRange" value="8.1"/>	  <!-- the maximum range of the sensor-->
        <param name="particles" value="60"/> <!--increased from 80-->
        
        <!--Update frequencies-->
        <param name="linearUpdate" value="0.25"/>     <!-- param name="linearUpdate" value="0.5"/-->
        <param name="angularUpdate" value="0.262"/>   <!--param name="angularUpdate" value="0.436"/-->
        <param name="temporalUpdate" value="-1.0"/>
        <param name="resampleThreshold" value="0.5"/>
        
        <!--Init map size-->
        <param name="xmin" value="-50.0"/>
        <param name="ymin" value="-50.0"/>
        <param name="xmax" value="50.0"/>
        <param name="ymax" value="50.0"/>
        <param name="delta" value="0.05"/>
        
        <!--All default-->
        <param name="sigma" value="0.05"/>
        <param name="kernelSize" value="1"/>
        <param name="lstep" value="0.05"/>
        <param name="astep" value="0.05"/>
        <param name="iterations" value="5"/>
        <param name="lsigma" value="0.075"/>
        <param name="ogain" value="3.0"/>
        <param name="lskip" value="0"/>
        <param name="srr" value="0.01"/>
        <param name="srt" value="0.02"/>
        <param name="str" value="0.01"/>
        <param name="stt" value="0.02"/>
        <param name="llsamplerange" value="0.01"/>
        <param name="llsamplestep" value="0.01"/>
        <param name="lasamplerange" value="0.005"/>
        <param name="lasamplestep" value="0.005"/>
    </node>
</launch>
